packages/ui/src/theme/utils.ts

```diff
@@ -34,20 +34,22 @@ function referenceValue(value: string) {
   return value;
 }
 
-export function cssValue(token: DesignToken<{ value: unknown }>) {
+export function cssValue(token: DesignToken<any>) {
   const { value } = token;
   if (isString(value)) {
     return referenceValue(value);
   }
 
-  if (isObject(value)) {
-    if ('offsetX' in value) {
-      return SHADOW_PROPERTIES.map((property) =>
-        // lookup property against `token` first for custom non-nested value, then lookup
-        // property against `value` for design token value, default to empty string
-        referenceValue(token[property] ?? value[property] ?? '')
-      ).join(' ');
-    }
+  if (isShadowToken(value) && isObject(value)) {
+    return SHADOW_PROPERTIES.map((property) => {
+      // lookup property against `token` first for custom non-nested value, then lookup
+      // property against `value` for design token value, default to empty string
+      if (isShadowToken(token) && isObject(token)) {
+        return referenceValue(token[property] ?? '');
+      } else {
+        return referenceValue(value[property] ?? '');
+      }
```

Comment:
These two conditions look the same to me and line 47 reads the same as line 43 or am i missing something 😄

Reply:
Yea I wonder if there is a cleaner way to write this code in a way that makes TS happy...

Reply:
I re-did this code a bit to be a bit more readable I hope. Take a look.

Reply:
Just throwing this out there but we should be able to remove `isObject` from the `if` blocks since we use `isObject` in `isShadowToken`. From there think we could probably just reduce the return to:
```
return referenceValue((isShadowToken(token) ? token[property] : value[property]) ?? '');
```

Reply:
Looks like we were mostly thinking the same thing 💯 Left [a question](https://github.com/aws-amplify/amplify-ui/pull/3312#discussion_r1072980384) on the new code, and still thinking we can remove the `isObject` from the first `if` block but looks much better imo